state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -340,10 +337,18 @@ pub mod pallet {
 	pub type HistoricalEarnedFees<T: Config> =
 		StorageDoubleMap<_, Identity, T::AccountId, Twox64Concat, Asset, AssetAmount, ValueQuery>;
 
+	#[pallet::storage]
+	pub type OrderModified<T: Config> =
+		StorageDoubleMap<_, Identity, T::AccountId, Twox64Concat, OrderId, u32, ValueQuery>;
```

Comment:
I had an idea for a better approach that storing and updating this in the pools pallet: We can store the counts in the Flip pallet instead. 

We need an enum that 'indexes' the call info, to be used as a key:

```rust
pub enum CallInfo {
	PoolOrderUpdate(AccountId, Asset),
	// Maybe other things that we want to throttle
}
```

Then a storage item in the Flip pallet:

```rust
pub type CallCounter<T: Config> =
	StorageMap<_, TwoX64Concat, CallInfo, u32, ValueQuery>
```

And we can update this storage in `correct_and_deposit_fee`.

Reply:
> How do the calls get pushed into the FLIP pallet?

The method that takes the fees into escrow has access to the call. We can map this to an internal `CallInfo` type, which contains just enough information to index the calls by 'category' that we want to restrict (for example `CallInfo::LimitOrder(AccountId, Asset)`) and then store that in the pallet.

It would require a trait similar to `WaiveFees::should_waive_fees` that is implemented at the runtime level and does the mapping.

Reply:
> We definitely want to include set_limit_order and set_range_order here too - not just update, I can only see set_limit_orders on mainnet, checked about 20 blocks for an update extrinsic, and couldn't find one.

Sure. The naming wasn't meant to imply that it would only include update_limit_order calls. The LimitOrderUpdate (or CallInfo::LimitOrder) could be used to index both set_limit_order and update_limit_order under the same 'bucket' of things.